opennms-doc/guide-admin/src/asciidoc/text/kafka-producer/kafka-producer.adoc

```diff
@@ -0,0 +1,220 @@
+// Allow GitHub image rendering
+:imagesdir: ../../images
+
+=== Overview
+
+The _Kafka Producer feature_ allows events, alarms and nodes from _{opennms-product-name}_ to be forwarded to _Kafka_.
+
+These objects are stored in different topics and the payloads are encoded using link:https://developers.google.com/protocol-buffers/[Google Protocol Buffers (GPB)].
+See `opennms-kafka-producer.proto` in the corresponding source distribution for the model definitions.
+
+==== Events
+
+The _Kafka Producer_ listens for all events on the event bus and forwards these to a _Kafka_ topic.
+The records are keyed by event _UEI_ and contain a _GPB_ encoded model of the event.
+
+By default, all events are forwarded to a topic named `events`.
+
+The name of the topic used can be configured, and an optional filtering expression can be set to help control which events are sent to the topic.
+
+==== Alarms
+
+The _Kafka Producer_ listens for changes made to the current set of alarms and forwards the resulting alarms to a _Kafka_ topic.
+The records are keyed by alarm reduction key and contain a _GPB_ encoded model of the alarm.
```

Comment:
> ... contain a GPB encoded model of the alarm

Question: The records contain a GPB encoded alarm or a node?

Reply:
In the alarms topic, it is a GPB encoded alarm.